src/server.cpp

```diff
@@ -2492,9 +2492,10 @@ void Server::SendBlocks(float dtime)
 	// Sort.
 	// Lowest priority number comes first.
 	// Lowest is most important.
-	std::sort(queue.begin(), queue.end());
-
-	ClientInterface::AutoLock clientlock(m_clients);
+	if (unique_clients > 1) {
+		// for a single client queue already sorted by dist
```

Comment:
in `GetNextBlocks` the sphere center is the predicted player position, while the distance used for sorting is from the camera. so this assumption doesn't work.

Reply:
In the first version (not published) the comment said "mostly sorted correctly", which is true.
The correct sorting is not important, just that it's mostly right.
I'll maintain the sorting here is not needed.

Reply:
Wait...
You had me confused with player-prediction. It's not relevant for this discussion.
Sorting the vector will sort by distance (priority in PrioritySortedBlockTransfer, which is set to d in getNextBlocks.)
Blocks are iterated in getNextBlock by unumerating over d, and blocks will be added to the vector in order of d, which is the right sort order (for each single client).

Reply:
Note that that wasn't case before 454483f